conn.go

```diff
@@ -35,12 +35,14 @@ func (c *Conn) RemoteAddr() net.Addr {
 	}
 }
 
+// TODO: is it okay to cast c.Conn().RemotePeer() into a multiaddr? might be "user input"
 func (c *Conn) RemoteMultiaddr() ma.Multiaddr {
-	a, err := ma.NewMultiaddr(fmt.Sprintf("/ipfs/%s/p2p-circuit/ipfs/%s", c.Conn().RemotePeer().Pretty(), c.remote.ID.Pretty()))
-	if err != nil {
-		panic(err)
-	}
-	return a
+	proto := ma.ProtocolWithCode(ma.P_P2P).Name
+	peerid := c.Conn().RemotePeer().Pretty()
+	p2paddr := ma.StringCast(fmt.Sprintf("/%s/%s", proto, peerid))
+
+	circaddr := ma.Cast(ma.CodeToVarint(P_CIRCUIT))
+	return p2paddr.Encapsulate(circaddr)
```

Comment:
Isn't this a little inconsistent? The addr we get can't be combined with an ID to be dialable -- instead it would need to be parsed.

Reply:
More of a nit though, we _need_ the peer part to differentiate connections; otherwise we'd just report `/p2p-circuit`.
But maybe that's what we want with the new semantics?

Reply:
> Isn't this a little inconsistent? The addr we get can't be combined with an ID to be dialable -- instead it would need to be parsed.

Not sure what you mean -- you'll get `/p2p/QmRelay/p2p-circuit`, and you can do the following to get a full peer address:
```go
peeraddr := ma.StringCast("/p2p/QmFoo")
conn.RemoteMultiaddr().Encapsulate(peeraddr)
```

> More of a nit though, we need the peer part to differentiate connections; otherwise we'd just report /p2p-circuit.  But maybe that's what we want with the new semantics?

Yeah I think we want to stop treating the underlying transport connection as something that's unique to a peer. (cc @Stebalien)

Reply:
Ok, I read that as `/p2p/QmRelay/p2p-circult/p2p/QmPeer`.
Getting back just the relay + p2p-circuit as transport is fine and consistent.